src/qt/pivx/settings/settingsexportcsv.cpp

```diff
@@ -0,0 +1,212 @@
+                break;
+            }
+            case SortTx::DATE_DESC:{
+                columnIndex = TransactionTableModel::Date;
+                break;
+            }
+            case SortTx::AMOUNT_ASC:{
+                columnIndex = TransactionTableModel::Amount;
+                order = Qt::AscendingOrder;
+                break;
+            }
+            case SortTx::AMOUNT_DESC:{
+                columnIndex = TransactionTableModel::Amount;
+                break;
+            }
+
+        }
+        txFilter->sort(columnIndex, order);
+
+        // name, column, role
+        writer.setModel(txFilter);
+        writer.addColumn(tr("Confirmed"), 0, TransactionTableModel::ConfirmedRole);
+        if (walletModel->haveWatchOnly())
+            writer.addColumn(tr("Watch-only"), TransactionTableModel::Watchonly);
+        writer.addColumn(tr("Date"), 0, TransactionTableModel::DateRole);
+        writer.addColumn(tr("Type"), TransactionTableModel::Type, Qt::EditRole);
+        writer.addColumn(tr("Label"), 0, TransactionTableModel::LabelRole);
+        writer.addColumn(tr("Address"), 0, TransactionTableModel::AddressRole);
+        writer.addColumn(BitcoinUnits::getAmountColumnTitle(walletModel->getOptionsModel()->getDisplayUnit()), 0, TransactionTableModel::FormattedAmountRole);
+        writer.addColumn(tr("ID"), 0, TransactionTableModel::TxIDRole);
+        fExport = writer.write();
+    }
+
+    if (fExport) {
+        inform(tr("Exporting Successful\nThe transaction history was successfully saved to %1.").arg(filename));
+    } else {
+        inform(tr("Exporting Failed\nThere was an error trying to save the transaction history to %1.").arg(filename));
+    }
+}
+
+void SettingsExportCSV::onExportAddressesClicked()
+{
+    if (filenameAddressBook.isNull()) {
+        inform(tr("Please select a folder to export the csv file first."));
+        return;
+    }
+
+    bool fExport = false;
+    if (walletModel) {
+
+        if (!addressFilter) {
+            addressFilter = new QSortFilterProxyModel(this);
+            addressFilter->setSourceModel(walletModel->getAddressTableModel());
+            addressFilter->setDynamicSortFilter(true);
+            addressFilter->setSortCaseSensitivity(Qt::CaseInsensitive);
+            addressFilter->setFilterCaseSensitivity(Qt::CaseInsensitive);
+
+            addressFilter->setFilterRole(AddressTableModel::TypeRole);
+            // Quick and dirty filter
+            switch (ui->comboBoxSortAddressType->itemData(ui->comboBoxSortAddressType->currentIndex()).toInt()) {
+                case 0:
+                    // Receive filter
+                    addressFilter->setFilterFixedString(AddressTableModel::Receive);
+                    break;
+                case 1:
+                default:
+                    // Send filter
+                    addressFilter->setFilterFixedString(AddressTableModel::Send);
+                    break;
+            }
```

Comment:
These lines should go outside `if (!addressFilter) {}`, otherwise the filter won't change anymore after being set the first time.

Reply:
yep, good catch.